### PR TITLE
adding unit tests

### PR DESCRIPTION
**Pull request by katiegoines**
### Instructions

**If this PR should not be merged upon approval for any reason, please submit as a DRAFT**

Which product(s) are affected by this PR (if applicable)?
- [ ] amplify-cli
- [ ] amplify-ui
- [ ] amplify-studio
- [ ] amplify-hosting
- [ ] amplify-libraries

Which platform(s) are affected by this PR (if applicable)?
- [ ] JS
- [ ] Swift
- [ ] Android
- [ ] Flutter
- [ ] React Native

**Please add the product(s)/platform(s) affected to the PR title**

#### Checks

- [ ] Does this PR conform to [the styleguide](https://github.com/aws-amplify/docs/blob/main/STYLEGUIDE.md)?

- [ ] Does this PR include filetypes other than markdown or images? Please add or update unit tests accordingly.

- [ ] Are any files being deleted with this PR? If so, have the needed redirects been created?

- [ ] Are all links in MDX files using the MDX link syntax rather than HTML link syntax? <br /> 
      _ref: MDX: `[link](https://docs.amplify.aws/)` 
            HTML: `<a href="https://docs.amplify.aws/">link</a>`_

### When this PR is ready to merge, please check the box below
- [ ] Ready to merge

_By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license._

### CODE REVIEW

File: src/components/FeatureLists/__tests__/FeatureLists.test.tsx [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+        Deploy apps in Next.js, Nuxt.js, Gatsby, React, Vue, Angular (and more)
+        by simply connecting your Git repository.
+      </FeatureItem>
+    </FeatureList>
+  );
+
+  it('should render the FeatureList component', async () => {
+    render(featureListComponent);
+
+    const heading = await screen.findByRole('heading', { name: 'Deploy' });
```
Comment (hbuchel) [inline src/components/FeatureLists/__tests__/FeatureLists.test.tsx:34]: Not blocking, but you can combine the H2 expect in this findByRole with:

```
await screen.findByRole('heading', { name: 'Deploy', level: 2 });
```

**Review by hbuchel — APPROVED** [review]
LGTM, left a non-blocking comment. Also this bumps statements to over 80% now, thanks! 🎉

File: src/components/FeatureFlags/__tests__/FeatureFlags.test.tsx [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+            'Uses previous policy format which allows control plane access to AppSync',
+          defaultNewProject: false,
+          defaultExistingProject: true
+        }
+      ]
+    };
+
+    const component = (
+      <FeatureFlagSummary
+        key={`feature-flag-summary-${1}`}
```
Comment (jacoblogan) [inline src/components/FeatureFlags/__tests__/FeatureFlags.test.tsx:40]: nit: Can we remove this key prop? I'm not seeing why we need it here.

File: src/components/FeatureFlags/__tests__/FeatureFlags.test.tsx [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
```
Comment (jacoblogan) [inline src/components/FeatureFlags/__tests__/FeatureFlags.test.tsx:77]: Same as above, `key` seems unnecessary here.

**Review by jacoblogan — APPROVED** [review]

**Pull request merged**